### PR TITLE
fixed minor typos in documentation

### PR DESCRIPTION
**Pull request by jameslamb**
Had a few minutes to kill today, so I ran `aspell check` over the `docs/` folder here. Please consider this PR to fix a few minor typos.

### CODE REVIEW

File: src/treelearner/gpu_tree_learner.cpp [hunk 1]
```diff
@@ -538,7 +538,7 @@ void GPUTreeLearner::AllocateGPUMemory() {
   }
   // data transfer time
   std::chrono::duration<double, std::milli> end_time = std::chrono::steady_clock::now() - start_time;
-  Log::Info("%d dense feature groups (%.2f MB) transfered to GPU in %f secs. %d sparse feature groups", 
+  Log::Info("%d dense feature groups (%.2f MB) transferred to GPU in %f secs. %d sparse feature groups",
```
Comment (jameslamb) [inline src/treelearner/gpu_tree_learner.cpp]: @Laurae2 good call. This is the only one I found (with `git grep transfered`)

**Comment by jameslamb** [comment]
The only thing that failed on my Travis CI build is the `check-docs` task, looks like it's because one of the links is down:

```
Real URL   https://www.kaggle.com/c/allstate-claims-severity/discussion/24520
Check time 2.229 seconds
Size       89B
Result     Error: 502 Bad Gateway
```

I can get there in the browser, so guessing it was a temporary outage at Kaggle. Going to manually trigger a rebuild on the task

File: .gitignore [hunk 2]
```diff
@@ -390,3 +390,6 @@ R-package/src/src/
 lightgbm_r/*
 lightgbm*.tar.gz
 lightgbm.Rcheck/
+
+# Files generate by aspell
```
Comment (StrikerRUS) [inline .gitignore]: "generate" -> "generate**d**"?

Comment (jameslamb) [reply .gitignore]: ha yes good catch, will fix right now

Comment (jameslamb) [reply .gitignore]: fixed

**Pull request merged**